frontend/packages/text-editor/src/TextRow.tsx

```diff
@@ -115,19 +108,22 @@ export const TextRow = ({
         );
       })}
       <TableCell>
-        <ButtonContainer>
+        <ButtonContainer className={classes.buttonGroup}>
           {textIdEditOpen ? (
-            <div>
-              <LegacyTextField
-                aria-label={'tekst key edit'}
-                isValid={!keyError}
-                value={textIdValue}
-                type='text'
-                onBlur={handleTextIdBlur}
-                onChange={handleTextIdChange}
-              />
-              {keyError ? <ErrorMessage role='alertdialog'>{keyError}</ErrorMessage> : null}
-            </div>
+            <FormField
+              value={textIdValue}
+              onChange={handleTextIdChange}
+              renderField={({ fieldProps }) => (
+                <Textfield
+                  {...fieldProps}
+                  aria-label={'text key edit'}
+                  error={keyError}
+                  onBlur={handleTextIdBlur}
```

Comment:
By checking for errors here instead, we can avoid invoking the blur method for no reason. 😊 
```suggestion
                  onBlur={keyError ? undefined : handleTextIdBlur}
```